src/mopup/__init__.py

```diff
@@ -74,25 +107,54 @@ def choicechanges(pkgfile: str) -> str:
     return dumpplist(dicts).decode()
 
 
-def main(interactive: bool, force: bool, minor_upgrade: bool, dry_run: bool) -> None:
+def main(
+    target_version: str | None,
+    interactive: bool,
+    force: bool,
+    minor_upgrade: bool,
+    dry_run: bool,
+) -> None:
     """Do an update."""
     _ensure_sudo_if_needed(dry_run)
 
     this_mac_ver = tuple(map(int, mac_ver()[0].split(".")[:2]))
     ver = compile_re(r"(\d+)\.(\d+).(\d+)/")
     macpkg = compile_re(r"python-(\d+\.\d+\.\d+(?:(?:a|b|rc)\d+)?)-macosx?(\d+).pkg")
 
-    thismajor, thisminor, thismicro, releaselevel, serial = version_info
-    level = {
-        "alpha": "a",
-        "beta": "b",
-        "candidate": "rc",
-        "final": "",
-    }[releaselevel]
+    if target_version:
+        target_ver = Version(target_version)
+        for pkg, _short_name, version in _find_python_packages(target_ver):
+            try:
+                python_exe = _get_python_executable(pkg, version)
+                exact_version_str = _get_exact_version(python_exe)
+                if exact_version_str:
+                    thispkgver = Version(exact_version_str)
+                    break
+            except LookupError:
+                continue  # not every Python package contains executables
```

Comment:
This seems a *little* broad? Do you want to give it the same treatment as you did for `MopUpValueError`? I'm a little worried about errors getting swallowed here.

You may have noticed that the previous code was *extremely* light on exception handling, because given the following factors:

1. this isn't super comprehensively tested
2. I don't have much of a durable "API" contract from python.org for the relevant package files or file listing layouts either on the website or in the installer

I really want any exceptions to propagate in easily legible ways.